assets/ts/api/request.ts

```diff
@@ -22,15 +22,20 @@ export function request<dataType>({ endpoint, query, body, method, headers }: re
         headers: {
             Authorization: `Bearer ${auth}`,
             "Content-Type": "application/json",
-            ...headers,
         },
         body: body ? JSON.stringify(body) : undefined,
     }).then((response) => {
-        return response.json();
+        if (response.ok) {
+            return response.json();
+        }
+        if (response.status === 403) {
+            navigateTo("/login");
+            throw new Error("Recieved '403 Forbidden' response from Spotify");
+        }
+        return response.text();
     }).then<dataType>((data) => {
         if (data.error) {
-            // throw for .catch
-            throw data.error;
+            throw new Error(data.error);
```

Comment:
das würde ich zurückändern (`throw data.error`)
bei throw new Error() sollte man einen human-readable string übergeben
durch throw data.error erhält der weitere code das error-object und kann es weiter verarbeiten

Reply:
data.error ist ein string. Wird bei `throw data.error` data.error wirklich automatisch in einem `Error` Objekt ge-wrapped?